instrumentation/net/http/otelhttp/config.go

```diff
@@ -37,6 +38,7 @@ type config struct {
 	Meter             metric.Meter
 	Propagators       propagation.TextMapPropagator
 	SpanStartOptions  []trace.SpanStartOption
+	MetricAttributes  []func(string, *http.Request, int) []attribute.KeyValue
```

Comment:
```suggestion
	MetricAttributeFns  []func(string, *http.Request, int) []attribute.KeyValue
```

The name should make clear that it is storing functions that will produce metric attributes and not the attributes themselves.